google/cloud/bigtable/admin/integration_tests/table_admin_integration_test.cc

```diff
@@ -274,64 +335,105 @@ TEST_F(TableAdminIntegrationTest, WaitForConsistencyCheck) {
   CreateCells(table, created_cells);
 
   // Create a consistency token after modifying the table.
+  auto consistency_token_resp = client_.GenerateConsistencyToken(table_name);
+  ASSERT_STATUS_OK(consistency_token_resp);
   auto consistency_token =
-      table_admin.GenerateConsistencyToken(random_table_id);
-  ASSERT_STATUS_OK(consistency_token);
+      std::move(*consistency_token_resp->mutable_consistency_token());
+
+  // A retry loop that checks if the data replication has completed.
```

Comment:
For a future PR: this is the kind of stuff that, while not hard to implement, it is tedious for customers to implement.  Maybe we should have a standalone function?  Something like:

```cc
namespace google::cloud::bigtable_admin {
future<Status> AsyncWaitForConsistency(BlahClient client, std::string consistency_token);
future<Status> AsyncWaitForConsistency(BlahClient client, std::string consistency_token, RetryPolicy retry, BackoffPolicy backoff);
}
```

Reply:
And FWIW, there is a lot of untested / unused code in your sample loop.

Reply:
Ack and I agree. We will keep `TableAdmin::WaitForConsistency()` for as long as we keep `TableAdmin`... but if we are ever going to deprecate `TableAdmin`, we need a useful alternative (like the one you propose).

It is true that the loop is not tested. Maybe it is already too complicated. But I am going to leave it as is. I think tests for all the behaviors of such a loop should be done in unit tests with mocks.

I wouldn't say there is "unused" code. If the loop fails for whatever reason, we want to have a more informative error as to why.